ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -183,7 +237,8 @@ void replicate(List<List<RemoteReplicaInfo>> replicasToReplicate) {
 
       List<RemoteReplicaInfo> activeReplicasPerNode = new ArrayList<RemoteReplicaInfo>();
       for (RemoteReplicaInfo remoteReplicaInfo : replicasToReplicatePerNode) {
-        if (!remoteReplicaInfo.getReplicaId().isDown()) {
+        ReplicaId replicaId = remoteReplicaInfo.getReplicaId();
+        if (!replicationDisabledPartitions.contains(replicaId.getPartitionId()) && !replicaId.isDown()) {
```

Comment:
Are there reasons why atomicity is desirable or important here? The contract is that replication will be disabled/enabled eventually (in the next cycle once things are added/removed from the map).